src/qibocal/protocols/characterization/randomized_benchmarking/standard_rb.py

```
@@ -192,71 +168,70 @@ def _acquisition(
         RBData: The depths, samples and ground state probability of each experiment in the scan.
     """
 
+    backend = GlobalBackend()
     # For simulations, a noise model can be added.
     noise_model = None
-    if params.noise_model:
-        # FIXME implement this check outside acquisition
-        if platform and platform.name != "dummy":
+    if params.noise_model is not None:
+        if backend.name == "qibolab":
             raise_error(
-                NotImplementedError,
-                f"Backend qibolab ({platform}) does not perform noise models simulation.",
-            )
-        elif platform:
-            log.warning(
-                (
-                    "Backend qibolab (%s) does not perform noise models simulation. "
-                    "Setting backend to ``NumpyBackend`` instead."
-                ),
-                platform.name,
+                ValueError,
+                "Backend qibolab (%s) does not perform noise models simulation. "
+                "Setting backend to ``NumpyBackend`` instead.",
```

If you `raise_error` instead of just a `log.warning` the code will stop instead of "Setting backend to ``NumpyBackend`` instead.". We should decide which one of the two to use. I would use the warning and set the backend to `numpy`.

---

I should have changed it, but I would rather leave the error or eventually ignore the noise model, if it the qibolab backend is set up. This is because the user expects the qibolab backend to run on hardware, so returning simulated data is unfair.

---

I am also fine with the error.
We just need to remove L179.